services-examples-parent/services-examples/src/main/java/io/scalecube/services/examples/exceptions/ExceptionMapperExample.java

```diff
@@ -31,24 +40,34 @@ public static void main(String[] args) throws InterruptedException {
 
     final Address address1 = ms1.discovery().address();
 
+    ServiceFactory serviceFactory =
+        new ServiceFactory() {
+          @Override
+          public Collection<ServiceDefinition> getServiceDefinitions() {
+            ServiceDefinition serviceA = new ServiceDefinition(ServiceA.class);
+            ServiceDefinition serviceB = new ServiceDefinition(ServiceB.class);
+            return List.of(serviceA, serviceB);
+          }
+
+          @Override
+          public Mono<? extends Collection<ServiceInfo>> initializeServices(
+              MicroservicesContext microservices) {
+            ServiceCall call = microservices.serviceCall();
+            ServiceA serviceA =
+                call.errorMapper(new ServiceAClientErrorMapper()).api(ServiceA.class);
+            ServiceB serviceB = new ServiceBImpl(serviceA);
+            ServiceInfo serviceInfoB = ServiceInfo.fromServiceInstance(serviceB).build();
+            return Mono.just(Collections.singletonList(serviceInfoB));
+          }
+        };
```

Comment:
Just realized how `ServiceFactory` api can shoot in the foot. In this example with service-a service-b in method `getServiceDefinitions` were returned two services, in method `initializeServices` was returned one service. That's ok that service-b injects service-a, but i imagine scenario with N services and now client has to maintain correlation bewtween what's returned from `getServiceDefinitions` and `initializeServices`. Don't have solution for this, but to me it looks like a big problem.

Reply:
There are only two ways. 

1. We create instances of services from ServiceDefinition
2. 

It is assumed that ServiceFactory will delegate its task to the DI framework and there will be no troubles. But they are possible. I will think about it.